### PR TITLE
fix: broken link

### LINKED ISSUE #1064: Contributor's Guide page not found
The contributor's guide link is broken
https://zio.dev/about/contributing in the README

### PR DESCRIPTION
**Pull request by ezhil56x**
/claim #1064
Closes #1064

### CODE REVIEW

**Comment by ezhil56x** [comment]
@jdegoes 
This issue is similar to [zio/zio-schema/644](https://github.com/zio/zio-schema/pull/644)
Can I get a review on [zio/zio-schema/660](https://github.com/zio/zio-schema/pull/660) as well?
Thanks!

**Comment by fsvehla** [comment]
```
[error] java.lang.RuntimeException: The README.md file is not up-to-date!
[error] Please run `sbt docs/generateReadme` and commit new changes.
[error] 	at scala.sys.package$.error(package.scala:30)
```

**Comment by ezhil56x** [comment]
> ```
> [error] java.lang.RuntimeException: The README.md file is not up-to-date!
> [error] Please run `sbt docs/generateReadme` and commit new changes.
> [error] 	at scala.sys.package$.error(package.scala:30)
> ```
@fsvehla 
https://github.com/zio/zio-schema/pull/644
In the above PR, it was suggested to remove `sbt docs/generateReadme` from CI. So I have removed it.

File: .github/workflows/site.yml [hunk 1]
```diff
@@ -75,8 +75,6 @@ jobs:
         distribution: temurin
         java-version: 17
         check-latest: true
-    - name: Generate Readme
-      run: sbt  docs/generateReadme
```
Comment (guizmaii) [inline .github/workflows/site.yml:79]: Not sure that was a good idea to remove that from the CI 🤔

**Pull request merged**